swig/cpp/tests/changes.cpp

```diff
@@ -0,0 +1,219 @@
+#include <iostream>
+#include <memory>
+#include <cassert>
+#include <cstring>
+#include <unistd.h>
+
+#include "Session.h"
+
+#define MAX_LEN 100
+
+using namespace std;
+
+const char *module_name = "swig-test";
+const int LOW_BOUND = 10;
+const int HIGH_BOUND = 20;
+const char *xpath_if_fmt = "/swig-test:cpp-operations/test-get[name='%s']/%s";
+
+char *
+get_test_name(int i)
+{
+    char buf[MAX_LEN] = {0};
+    sprintf(buf, "test-cpp-%d", i);
+    return strdup(buf);
+}
+
+const char *
+get_xpath(char *test_name, char *node_name)
+{
+    char buf[MAX_LEN];
+    sprintf(buf, xpath_if_fmt, test_name, node_name);
+    return strdup(buf);
+}
+
+class NOPCb: public Callback {
+public:
+    int module_change(S_Session sess, const char *module_name, sr_notif_event_t event, void *private_ctx)
+        {
+            return SR_ERR_OK;
+        }
+};
+
+void init_test(S_Session sess)
+{
+    S_Subscribe subs(new Subscribe(sess));
+    S_Callback cb(new NOPCb());
+
+    subs->module_change_subscribe(module_name, cb, NULL, 0, SR_SUBSCR_DEFAULT | SR_SUBSCR_APPLY_ONLY);
+
+    for (int i = LOW_BOUND; i < HIGH_BOUND; i++) {
+        const char *xpath = (char*)get_xpath(get_test_name(i), (char *) "number");
```

Comment:
Again, this is a matter of C++-ish preferences. These casts are not needed (or can be made not needed) especially if one follows the C++ idioms instead of the C ones.

```C++
const auto xpath = prepare_test_xpath(get_test_name(1), "number");
```